lib/dashing/app.rb

```diff
@@ -70,13 +70,23 @@ def authenticated?(token)
   redirect "/" + dashboard
 end
 
+
 get '/events', provides: 'text/event-stream' do
   protected!
   response.headers['X-Accel-Buffering'] = 'no' # Disable buffering for nginx
-  stream :keep_open do |out|
-    settings.connections << out
+  stream do |out|
     out << latest_events
-    out.callback { settings.connections.delete(out) }
+    settings.connections << connection = {out: out, mutex: Mutex.new, terminated: false}
+    terminated = false
+
+    loop do
+      connection[:mutex].synchronize do
+        terminated = true if connection[:terminated]
+      end
+      break if terminated
+    end
+
```

Comment:
I don't think so. I am not sure, but I think this is because Sinatra doesn't properly implement Rack hijack.

The while loop stuff comes from this: http://tenderlovemaking.com/2012/07/30/is-it-live.html